src/client/dfs/dfs_internal.h

```diff
@@ -59,6 +59,12 @@ dfs_get_sb_layout(daos_key_t *dkey, daos_iod_t *iods[], int *akey_count,
 void
 dfs_free_sb_layout(daos_iod_t *iods[]);
 
+/* as dfs_open() but takes a stbuf */
+int
+dfs_open2(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
+	  int flags, daos_oclass_id_t cid, daos_size_t chunk_size,
+	  const char *value, struct stat *stbuf, dfs_obj_t **obj);
```

Comment:
would prefer if stbuf was last argument

Reply:
Done.  obj is also an out arg so I'd left that one last.